src/runtime/EnumerateObject.cpp

```diff
@@ -222,8 +223,14 @@ void EnumerateObjectWithIteration::executeEnumeration(ExecutionState& state, Enc
         }
         m_hiddenClassChain.push_back(proto->structure());
         proto = proto->getPrototypeObject(state);
+
+        // v8 throw exception when there is too many things on prototype chain
+        if (UNLIKELY(m_hiddenClassChain.size() > 1024 * 128)) {
```

Comment:
What is this number `1024*128`?
v8 benchmark arbitrarily sets this threshold number to check huge prototype chains?

Reply:
enabled v8 test has infinity prototype chain with ProxyObject.
ordinary object prevents infinity prototype when setting __proto__,
but there is no limitation for ProxyObject.
1024 * 128 is practical limit